clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4166,10 +4167,22 @@ public DescribeUserScramCredentialsResult describeUserScramCredentials(List<Stri
         Call call = new Call("describeUserScramCredentials", calcDeadlineMs(now, options.timeoutMs()),
                 new LeastLoadedNodeProvider()) {
             @Override
-            public DescribeUserScramCredentialsRequest.Builder createRequest(int timeoutMs) {
-                return new DescribeUserScramCredentialsRequest.Builder(
-                        new DescribeUserScramCredentialsRequestData().setUsers(users.stream().map(user ->
-                                new DescribeUserScramCredentialsRequestData.UserName().setName(user)).collect(Collectors.toList())));
+            public DescribeUserScramCredentialsRequest.Builder createRequest(final int timeoutMs) {
+                final DescribeUserScramCredentialsRequestData requestData = new DescribeUserScramCredentialsRequestData();
+
+                if (users != null && !users.isEmpty()) {
+                    final List<UserName> userNames = new ArrayList<>(users.size());
+
+                    for (final String user : users) {
+                        if (user != null) {
+                            userNames.add(new UserName().setName(user));
+                        }
+                    }
+
+                    requestData.setUsers(userNames);
```

Comment:
Not really, if we keep all checks in place it will only be a little shorter but definitely less efficient:

```java
final DescribeUserScramCredentialsRequestData requestData = new DescribeUserScramCredentialsRequestData();

if (users != null && !users.isEmpty()) {
    requestData.setUsers(users.stream()
        .filter(Objects::nonNull)
        .map(it -> new DescribeUserScramCredentialsRequestData.UserName().setName(it))
        .collect(Collectors.toList())
    );
}

return new DescribeUserScramCredentialsRequest.Builder(requestData);
```

Reply:
We tend to use the steam api for such small transformations but I don't feel strong about this.

Reply:
I have none either, for me Java is always just clunky no matter which API you use. A normal loop is just faster, that's all I can say, but I doubt that it matters at all.